packages/playwright-core/src/server/fetch.ts

```diff
@@ -583,7 +583,7 @@ function serializePostData(params: channels.APIRequestContextFetchParams, header
     }
     headers['content-type'] ??= formData.contentTypeHeader();
     return formData.finish();
-  } else if (params.postData) {
+  } else if (isString(params.postData)) {
```

Comment:
'Twas a flawed "consistency" line of thought on my behalf: I thought I was being consistent with some other [existing `isString`](https://github.com/microsoft/playwright/blob/7d924470d397975a74a19184c136b3573a974e13/packages/playwright-core/src/client/network.ts) usage in thinking it was some stylistic thing, but it turns out (as you point out) in this context functionally worse since we are deciding between `string | undefined`, not `string | object | etc.`.  Oops! Thanks!